src/index.tsx

```diff
@@ -129,45 +127,58 @@ export function useSpeechRecognition(options: SpeechRecognitionOptions = default
   );
 
   const onRecognitionDisconnect = useCallback(() => {
-    dispatch(setStatus(SpeechRecognitionStatus.STOPPED));
     if (pauseAfterDisconnect) {
-      dispatch(setListening(false));
+      dispatch(setStatus(SpeechRecognitionStatus.STOPPED));
     } else if (recognition) {
       if (recognition.continuous) {
         startListening();
       } else {
-        dispatch(setListening(false));
+        dispatch(setStatus(SpeechRecognitionStatus.STOPPED));
       }
     }
     dispatch(setPauseAfterDisconnect(false));
   }, [pauseAfterDisconnect, recognition, startListening]);
 
   const onRecognitionError = useCallback(({ error, message }) => {
-    console.log("Speech recognition error detected: " + error);
-    console.log("Additional information: " + message);
+    dispatch(setStatus(SpeechRecognitionStatus.ERROR));
+    dispatch(setError(`${error}: ${message}`));
   }, []);
 
   useEffect(() => {
-    if (recognition && !listening) {
+    if (recognition) {
       recognition.continuous = options.continuous !== false;
       recognition.interimResults = options.interimResults;
       recognition.onresult = updateTranscript;
       recognition.onend = onRecognitionDisconnect;
       recognition.onerror = onRecognitionError;
```

Comment:
can this code be moved to the initialization of the recognition object on line 52 above

Reply:
I think this makes the effect unnecessary?

Reply:
It would work but the dependencies then get messed up. The `recognition` `useMemo` depends on the callback functions' existence but the callbacks depend on `recognition`. It doesn't work out to combine them.